### PR TITLE
Fix vscode api enum

### PR DESCRIPTION
**Pull request by CGNonofr**
fix for https://github.com/TypeFox/monaco-languageclient/issues/332

### CODE REVIEW

**Comment by CGNonofr** [comment]
@kaisalmen I think this fix is pretty urgent

**Review by kaisalmen — APPROVED** [review]
👍 import type is a very good idea. Just have a look at my question. Thanks.

File: client/src/vscode-api.ts [hunk 1]
```diff
@@ -119,6 +119,14 @@ export function createVSCodeApi(servicesProvider: Services.Provider): typeof vsc
         }
     }
 
+    enum TextDocumentChangeReason {
```
[the remark this answers is not in the document]
Comment (CGNonofr) [reply client/src/vscode-api.ts:122]: The error was at list! `reason: isUndoing ? vscode.TextDocumentChangeReason.Undo : isRedoing ? vscode.TextDocumentChangeReason.Redo : undefined`

`vscode.TextDocumentChangeReason` wasn't defined so it was crashing

I hope I understood your question

[the remark this answers is not in the document]
Comment (kaisalmen) [reply client/src/vscode-api.ts:122]: Yes, you understood me. Thanks. 👍🙂 Just by looking at the PR without having deep knowledge of the project yet doesn't make the problem directly clear. That's why `import type` (did not know it and look it up) is very helpful here as it immediately makes the issue obvious.

**Comment by kaisalmen** [comment]
I will merge and release `0.18.1`.

File: client/src/vscode-api.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (CGNonofr) [reply client/src/vscode-api.ts:122]: :+1: The same way we need to re-implement all vscode.* interfaces, enums in typescript are transpiled into objects, and we need those at runtime

**Comment by CGNonofr** [comment]
> I will merge and release `0.18.1`.

you rocks!

**Pull request merged**